doc/developer/platform/formalism.md

```diff
@@ -194,3 +199,111 @@ A view introduces *constraints* on the capabilities and frontiers of its input a
     Outputs may need to be recovered in the case of failure, but also in other less dramatic query migration scenarios.
 
 These constraints are effected using the capability abstractions.
+The Compute layer acquires and maintains read capabilities for collections managed by it, and by the Storage layer.
+
+# Adapter
+
+The adapter layer translates SQL statements into commands for the Storage and Compute layers.
+
+The most significant differences between SQL statements and commands to the Storage and Compute layers are:
+1.  The absence of explicit timestamps in SQL statements.
+
+    A `SELECT` statement does not indicate *when* it should be run, or against which version of its input data.
+    The Adapter layer introduces timestamps to these commands, in a way that provides the appearance of sequential execution.
+
+2.  The use of user-defined and reused names in SQL statements rather than `GlobalId` identifiers.
+
+    SQL statements reference user-assigned names that may be rebound over time.
+    The meaning, and even the validity, of a query depends on the current association of user-defined name to `GlobalId`.
+    The Adapter layer maintains a map from user-assigned names to `GlobalId` identifiers, and introduces new ones as appropriate.
+
+Generally, SQL is "more ambiguous" than the Storage and Compute layers, and the Adapter layer must resolve that ambiguity.
+
+The SQL commands are a mix of DDL (data definition language) and DML (data manipulation language).
+In Materialize today, the DML commands are timestamped, and the DDL commands are largely not (although perhaps they should be).
+That DDL commands are not timestamped is a known potential source of all manner of apparent consistency issues.
+While it may be beneficial to think of Adapter's state as a pTVC, changes to its state are not meant to cascade through established views definitions.
+
+Commands acknowledged by the Adapter layer are durably recorded in a total order.
+Any user can rely on all future behavior of the system reflecting any acknowledged command.
```

Comment:
This sounds like linearizability, which we don't have for non-tables, so I assume this comment is intended rather than current behavior?

Reply:
Yes, aspirational. I think we do still aspire to this, but lmk if that is wrong in any way!

Reply:
Nope, sgtm as is if aspirational.